core/chaincode/handler_test.go

```diff
@@ -1106,6 +1259,208 @@ var _ = Describe("Handler", func() {
 		})
 	})
 
+	Describe("HandleGetStateMultipleKeys", func() {
+		var (
+			incomingMessage  *pb.ChaincodeMessage
+			request          *pb.GetStateMultiple
+			expectedResponse *pb.ChaincodeMessage
+			response         [][]byte
+			respProto        *pb.GetStateMultipleResult
+			respPayload      []byte
+		)
+
+		BeforeEach(func() {
+			request = &pb.GetStateMultiple{
+				Keys: []string{"get-state-key1", "get-state-key2", "get-state-key3"},
+			}
+			payload, err := proto.Marshal(request)
+			Expect(err).NotTo(HaveOccurred())
+
+			incomingMessage = &pb.ChaincodeMessage{
+				Type:      pb.ChaincodeMessage_GET_STATE_MULTIPLE,
+				Payload:   payload,
+				Txid:      "tx-id",
+				ChannelId: "channel-id",
+			}
+
+			expectedResponse = &pb.ChaincodeMessage{
+				Type:      pb.ChaincodeMessage_RESPONSE,
+				Txid:      "tx-id",
+				ChannelId: "channel-id",
+			}
+
+			response = [][]byte{[]byte("my-value1"), nil, []byte("my-value3")}
+			respProto = &pb.GetStateMultipleResult{
+				Values: response,
+			}
+			respPayload, err = proto.Marshal(respProto)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		Context("when unmarshalling the request fails", func() {
+			BeforeEach(func() {
+				incomingMessage.Payload = []byte("this-is-a-bogus-payload")
+			})
+
+			It("returns an error", func() {
+				_, err := handler.HandleGetStateMultipleKeys(incomingMessage, txContext)
+				Expect(err).To(Not(BeNil()))
+				Expect(err.Error()).To(HavePrefix("unmarshal failed:"))
+			})
+		})
+
+		Context("when collection is set", func() {
+			BeforeEach(func() {
+				request.Collection = "collection-name"
+				payload, err := proto.Marshal(request)
+				Expect(err).NotTo(HaveOccurred())
+				incomingMessage.Payload = payload
+
+				fakeCollectionStore.RetrieveReadWritePermissionReturns(true, false, nil)
+				fakeTxSimulator.GetPrivateDataMultipleKeysReturns(response, nil)
+				expectedResponse.Payload = respPayload
+			})
+
+			It("calls GetPrivateData on the transaction simulator", func() {
```

Comment:
Improve the It message. Did you intend to mix HandleGetStateMultipleKeys() and GetPrivateDataMultipleKeysCallCount() below?

Same comment for multiple tests below, maybe copy/paste mistakes? Please double check them all...

Reply:
fixed